element_interactions/test/specs/test.e2e.js

```diff
@@ -0,0 +1,170 @@
+        let isEnabled = inputField.isEnabled();
+        // assert.equal(isEnabled, false, "Element not disabled")
+        assert.isFalse(await isEnabled, "Element is not disabled");
+
+        // Enable it
+        let enableDisableButton = $('//form[@id="input-example"]/button[@type="button"]')
+        await enableDisableButton.click();
+
+        // Verify the enable text
+        const enabledText = $('//p[text()="It\'s enabled!"]');
+        assert.equal(await enabledText.getText(), "It's enabled!", "Text doesn't match");
+
+        // Assert the input field is enabled
+        // assert.equal(isEnabled, true, "Input field not enabled")
+        isEnabled = inputField.isEnabled();
+        assert.isTrue(await isEnabled, "The field is still disabled!");
+
+        // assert.equal(enableDisableButton.getText(), 'Disable', "Input field not enabled" );
+
+
+    });
+
+    it('should upload a file', async () => {
+
+        await browser.url('http://the-internet.herokuapp.com/upload');
+
+
+        const chooseFileButton = $('#file-upload');
+        const submitFileButton = $('#file-submit');
+
+        // Upload file  
+
+        const filePath = 'C:\\Vention\\Vention_internship\\element_interactions\\test\\specs\\testFile.txt'
+        const remoteFilePath = await browser.uploadFile(filePath)
+
+        await chooseFileButton.setValue(remoteFilePath)
+        await submitFileButton.click()
+
+
+        // Verify that the file has been uploaded
+        const successMessage = $('//h3[text()="File Uploaded!"]');
+        await successMessage.waitForDisplayed();
+
+        // Verify that the uploaded file name matches the expected file name
+        const uploadedFileName = await $('//div[@id="uploaded-files"]').getText();
+        assert.equal(uploadedFileName, "testFile.txt", 'Uploaded file name does not match');
+    });
+
+    it('should verify iFrame content', async () => {
+        await browser.url('http://the-internet.herokuapp.com/frames');
+
+        // Switch to the iframe
+        const iframeLink = $("//a[text()='iFrame']");
+        await iframeLink.click();
+        const iframe = await $('iframe')
+        await browser.switchToFrame(iframe);
+
+        // Assert the text of element
+        const iframeTextLocator = $("//p");
+        const iframeText = await iframeTextLocator.getText();
+        const expectedText = 'Your content goes here.';
+        assert.equal(iframeText, expectedText, "Text doesn't match");
+    });
+
+
+    it.only('should download and verify file', async () => {
+
+        await browser.url(`https://the-internet.herokuapp.com/download`);
+        const links = await $$('//a');
+        const extensions = ['.jpg', '.txt', '.png', '.json'];
+        const allowedLinks = [];
+
+        for (const link of links) {
+            const name = await link.getText();
+            allowedLinks.push(name);
+        }
+
+
+        const filteredLinks = allowedLinks.filter(linkName => {
```

Comment:
filteredLinks will include not all allowed files
![image](https://github.com/ElenaZlatevaNikolova/Vention_internship/assets/98515302/3452a1c7-012f-4b63-b442-30ccd965d543)